telemetry/definitions/commonDefinitions.json

```diff
@@ -831,10 +831,23 @@
             "metadata": [
                 { "type": "result" },
                 { "type": "initialDeploy" },
+                { "type": "duration" },
                 { "type": "name", "required": false },
                 { "type": "framework", "required": false },
                 { "type": "xrayEnabled", "required": false },
-                { "type": "enhancedHealthEnabled", "required": false }
+                { "type": "enhancedHealthEnabled", "required": false },
+                { "type": "serviceType", "required": false },
+                { "type": "source" , "required": false}
+            ]
+        },
+        {
+            "name": "beanstalk_publishWizard",
+            "description": "Called when user completes the Elastic Beanstalk publish wizard",
+            "metadata": [
+                { "type": "result" },
+                { "type": "duration" },
```

Comment:
Maybe add `duration` as a base field to everything similar to `reason`?

Reply:
Do we see this getting potentially used for majority of our metrics in future? wondering for one-click actions like Copy/View?

Reply:
It would be reasonable for duration to be an optional base field. Such work would take place in a separate change though. (duration and result could be added to the base class in one pass)

Reply:
Sounds good. I'll merge this and add those in a separate PR